### ISSUE TITLE
exception within exception while deleting instance with no vif

### ISSUE THREAD

**Issue by anshprat** [Juniper/contrail-nova-vif-driver#18]
Hi,

We are getting the following exception while doing a delete operation. 
This particular instance is in a bad state since this does not have a port attached to it. Basically the deletes in this issue has been handled by

https://bugs.launchpad.net/nova/+bug/1306718

However when using the contrail driver, the above issue causes an exception which triggers another exception in log print

 ```python
Traceback (most recent call last):
   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 2315, in do_terminate_instance
     self._delete_instance(context, instance, bdms, quotas)
   File "/usr/lib/python2.7/dist-packages/nova/hooks.py", line 103, in inner
     rv = f(*args, **kwargs)
   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 2284, in _delete_instance
     quotas.rollback()
   File "/usr/lib/python2.7/dist-packages/nova/openstack/common/excutils.py", line 68, in __exit__
     six.reraise(self.type_, self.value, self.tb)
   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 2255, in _delete_instance
     self._shutdown_instance(context, db_inst, bdms)
   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 2197, in _shutdown_instance
     requested_networks)
   File "/usr/lib/python2.7/dist-packages/nova/openstack/common/excutils.py", line 68, in __exit__
     six.reraise(self.type_, self.value, self.tb)
   File "/usr/lib/python2.7/dist-packages/nova/compute/manager.py", line 2187, in _shutdown_instance
     block_device_info)
   File "/usr/lib/python2.7/dist-packages/nova/virt/libvirt/driver.py", line 963, in destroy
     destroy_disks)
   File "/usr/lib/python2.7/dist-packages/nova/virt/libvirt/driver.py", line 999, in cleanup
     self.unplug_vifs(instance, network_info, ignore_errors=True)
   File "/usr/lib/python2.7/dist-packages/nova/virt/libvirt/driver.py", line 867, in unplug_vifs
     self.vif_driver.unplug(instance, vif)
   File "/usr/lib/python2.7/dist-packages/nova_contrail_vif/contrailvif.py", line 197, in unplug
     pformat(instance.__dict__))))
 AttributeError: 'dict' object has no attribute '__dict__'
```

This above issue of instance being a dict type has been handled correctly elsewhere in the code:

https://github.com/Juniper/contrail-nova-vif-driver/blob/master/nova_contrail_vif/contrailvif.py#L253

            if isinstance(instance, dict):

Basically, we need to do the same even in the exception log printing.

 ```python
except Exception as e:
            from pprint import pformat
            LOG.error(_("Error in unplug: %s locals: %s instance %s"
                       %(str(e), pformat(locals()),
                         pformat(instance.__dict__))))
```

**Comment by pedro-r-marques** [comment]
Please file the bug report via bugs.launchpad.net/opencontrail.
